### PR TITLE
feat(client-control): set client_max_body_size dynamically

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: conf/config-default.yaml [hunk 1]
```diff
@@ -243,6 +243,7 @@ graphql:
   #cmd: ["ls", "-l"]
 
 plugins:                          # plugin list (sorted by priority)
+  - client-control             # priority: 22000
```
Comment (tokers) [inline conf/config-default.yaml:246]: Keep the comment with same indent?

File: docs/en/latest/plugins/client-control.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+---
+title: ext-plugin-pre-req
```
Comment (tokers) [inline docs/en/latest/plugins/client-control.md:2]: bad title.

File: docs/en/latest/plugins/client-control.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+- [**How To Enable**](#how-to-enable)
+- [**Test Plugin**](#test-plugin)
+- [**Disable Plugin**](#disable-plugin)
+
+## Name
+
+The `client-control` plugin dynamically controls the behavior of Nginx to
+handle the client request.
+
+This plugin requires APISIX running on the APISIX-OpenResty.
```
Comment (juzhiyuan) [inline docs/en/latest/plugins/client-control.md:37]: Do we need to add link for `APISIX-OpenResty`?

**Review by juzhiyuan — APPROVED** [review]
Logics are hard for me, only check the documentation :)

**Review by membphis — APPROVED** [review]

File: docs/en/latest/plugins/client-control.md (hunk 3 above)
Comment (spacewander) [reply docs/en/latest/plugins/client-control.md]: @juzhiyuan 
Updated.

File: t/plugin/client-control.t [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                    "uri": "/hello",
+                    "upstream": {
+                        "type": "roundrobin",
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        }
+                    },
+                    "plugins": {
+                        "client-control": {
+                            "max_body_size": 5
```
Comment (tokers) [inline t/plugin/client-control.t:169]: It seems that the configuration doesn't match the title (`0` means no limit).

**Review by tzssangglass — APPROVED** [review]

File: t/plugin/client-control.t (hunk 4 above)
Comment (spacewander) [reply t/plugin/client-control.t]: @tokers 
Updated.

**Review by tokers — APPROVED** [review]

**Pull request merged**